packages/@icon-magic/config-reader/src/schemas/config-schema.ts

```diff
@@ -3,14 +3,61 @@ const variantProperties = {
   path: { type: 'string', minLength: 1 }
 };
 
+const pluginProperties = {
+  name: { type: 'string' },
+  iterants: {
+    type: ['array'],
+    items: {
+      type: 'string'
+    }
+  },
+  params: ['object'],
+  writeToOutput: {
+    type: 'boolean',
+    default: true
+  }
+};
+
 const buildConfigProperties = {
-  outputPath: { type: ['string', 'null'] },
-  plugins: { type: ['array', 'null'] }
+  outputPath: {
+    type: ['string', 'null'],
+    minLength: 1
+  },
+  plugins: {
+    type: ['array', 'null'],
+    items: {
+      type: ['object'],
+      properties: pluginProperties,
+      additionalProperties: true
+    }
+  }
 };
 
 const distributeConfigProperties = {};
 
-const generateConfigProperties = {};
+const generateConfigProperties = {
+  outputPath: {
+    type: ['string', 'null'],
+    minLength: 1
+  },
+  types: {
+    type: ['array'],
+    items: {
+      type: ['object'],
+      properties: {
+        name: { type: 'string' },
+        plugins: {
+          type: ['array'],
+          items: {
+            type: ['object'],
+            properties: pluginProperties,
+            additionalProperties: true
```

Comment:
I'm on the fence about this about allowing additionalProperties

Reply:
removed

Reply:
ah wait actually, what about `fn: PluginFunctionType<T>;` ?

Reply:
Ok, so `fn` is going to be optional again, because if that isn't supplied, then we look under `build/plugins` and `generate/plugins` for a plugin of the same `name` passed in and override the rest of the plugin values from what is passed in the config.

Reply:
afaik I can't define `fn` as a property because function isn't a valid type? which is why I have `additionalProperties` to account for it since I don't define it in `pluginProperties`

Reply:
Ah ok, that makes sense. I did find this though: https://github.com/epoberezkin/ajv-keywords for `function`. We could do that, or keep additionalProperties as you have. Your call.

Reply:
Ok, I'd rather have `additionalProperties` than add a new package, and we're not using function types that often.